lib/irc/ClientPool.js

```diff
@@ -30,6 +30,33 @@ function ClientPool(ircBridge) {
     }
 }
 
+ClientPool.prototype.killAllClients = function() {
+    var domainList = Object.keys(this._virtualClients);
+    var killList = domainList.map((domain) => {
+        let clients = Object.keys(this._virtualClients[domain].nicks).map(
+            (nick) => this._virtualClients[domain].nicks[nick]
+        );
+
+        clients = clients.concat(
+            Object.keys(this._virtualClients[domain].userIds).map(
+                (userId) => this._virtualClients[domain].userIds[userId]
```

Comment:
I feel like there should be a better data structure here then if there is a genuine redundancy. Also killing a killed client is a nop.